src/components/shared/Post/RenderMarkdown.tsx

```diff
@@ -34,6 +60,24 @@ export default function RenderMarkdown({ content }: any) {
             </pre>
           );
         },
+        h1: ({ children }) => (
+          <HeadingWithAnchor level={1}>{children}</HeadingWithAnchor>
+        ),
+        h2: ({ children }) => (
+          <HeadingWithAnchor level={2}>{children}</HeadingWithAnchor>
+        ),
+        h3: ({ children }) => (
+          <HeadingWithAnchor level={3}>{children}</HeadingWithAnchor>
+        ),
+        h4: ({ children }) => (
+          <HeadingWithAnchor level={4}>{children}</HeadingWithAnchor>
+        ),
+        h5: ({ children }) => (
+          <HeadingWithAnchor level={5}>{children}</HeadingWithAnchor>
+        ),
+        h6: ({ children }) => (
+          <HeadingWithAnchor level={6}>{children}</HeadingWithAnchor>
+        ),
```

Comment:
This is perfect and the exact type of solution I would have gone with. Excellent